### PR TITLE
feat(scalasdk): Codegen for Views, #372

### PR DESCRIPTION
**Pull request by patriknw**
* and starting at scala-value-entity-customer-registry sample

Doesn't work yet, but sharing...

Refs #372

### CODE REVIEW

File: samples/scala-value-entity-customer-registry/project/plugins.sbt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+addSbtPlugin("com.akkaserverless" % "sbt-akkaserverless" % System.getProperty("akkaserverless-sdk.version", "0.7.1"))
```
Comment (patriknw) [inline samples/scala-value-entity-customer-registry/project/plugins.sbt:1]: `publishLocal` and then version with:
```
sbt -Dakkaserverless-sdk.version="0.7.1-23-7a6dfec4-SNAPSHOT"
```

File: samples/scala-value-entity-customer-registry/project/plugins.sbt [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "2.1.0")
```
[the remark this answers is not in the document]
Comment (raboof) [reply samples/scala-value-entity-customer-registry/project/plugins.sbt:2]: I didn't need it yet in the current scripted test, but when we refactor the 'service call' API's to use Akka gRPC I guess we should pull it in automatically

File: codegen/core/src/main/scala/com/lightbend/akkasls/codegen/ModelBuilder.scala [hunk 3]
```diff
@@ -297,6 +301,13 @@ object ModelBuilder {
             Some(ActionService(serviceName, commands))
           case ServiceType.SERVICE_TYPE_VIEW =>
             val methodDetails = methods.flatMap { method =>
+              println(
+                s"# ${serviceName.protoName} method ${method.getName} [${method
+                  .getOptions()
+                  .getExtension(com.akkaserverless.Annotations.method)
+                  .getView()
+                  .hasUpdate}] [${method.getOptions().getExtension(com.akkaserverless.Annotations.method).getView().hasQuery}]"
+              ) // FIXME
```
Comment (patriknw) [inline codegen/core/src/main/scala/com/lightbend/akkasls/codegen/ModelBuilder.scala:310]: Added `Annotations.registerAllExtensions`, but there is still something strange. Looking at the debug output from this, it seems like the CustomerByName is parsed twice, and second time it has false on the `hasUpdate`

```
[DEBUG] Looking at descriptor customer/view/customer_view.proto
# CustomerByName method UpdateCustomer [true] [false]
# CustomerByName method GetCustomers [false] [true]
# CustomerByEmail method UpdateCustomer [true] [false]
# CustomerByEmail method GetCustomer [false] [true]
# CustomerSummaryByName method GetCustomers [false] [true]
# CustomerSummaryByName method UpdateCustomer [true] [false]
# CustomersResponseByName method GetCustomers [false] [true]
# CustomersResponseByName method UpdateCustomer [true] [false]
[DEBUG] Raw value entity name:
[DEBUG] Raw replicated entity name:
[DEBUG] Looking at descriptor customer/action/customer_action.proto
[DEBUG] Raw value entity name:
[DEBUG] Raw replicated entity name:
[DEBUG] Looking at descriptor customer/api/customer_api.proto
[DEBUG] Raw value entity name:
[DEBUG] Raw replicated entity name:
[DEBUG] Looking at descriptor customer/domain/customer_domain.proto
[DEBUG] Raw value entity name: CustomerValueEntity
[DEBUG] Raw replicated entity name:
[DEBUG] Looking at descriptor customer/view/customer_view.proto
# CustomerByName method UpdateCustomer [false] [false]
# CustomerByName method GetCustomers [false] [false]
```

File: samples/scala-value-entity-customer-registry/build.sbt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+Compile / javacOptions ++= Seq("-Xlint:unchecked", "-Xlint:deprecation")
+
+Test / parallelExecution := false
+Test / testOptions += Tests.Argument("-oDF")
+Test / logBuffered := false
+
+run / fork := false
+Global / cancelable := false // ctrl-c
+
+Compile / compile := {
```
[the remark this answers is not in the document]
Comment (patriknw) [reply samples/scala-value-entity-customer-registry/build.sbt:33]: solved by #429

File: samples/scala-value-entity-customer-registry/project/plugins.sbt (hunk 2 above)
[the remark this answers is not in the document]
Comment (patriknw) [reply samples/scala-value-entity-customer-registry/project/plugins.sbt:2]: Without sbt-akka-grpc it doesn't generate anything (from sbt-akkaserverless)

File: samples/scala-value-entity-customer-registry/src/main/proto/customer/domain/customer_domain.proto [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// tag::declarations[]
+syntax = "proto3";
+
+package customer.domain;
+
+option java_outer_classname = "CustomerDomain";
```
Comment (patriknw) [inline samples/scala-value-entity-customer-registry/src/main/proto/customer/domain/customer_domain.proto:20]: The generated code in the sample is looking good, aside from that CustomerDomain is different for Scala, but I think @raboof is looking into that problem.

**Comment by patriknw** [comment]
I have rebased this. Still compilation error for the State references, but that is a know issue. I think we can review/merge this anyway.

**Review by octonato — APPROVED** [review]

**Comment by octonato** [comment]
I'm merging this

**Pull request merged**